app/controllers/carts_controller.rb

```diff
@@ -1,20 +1,36 @@
 class CartsController < ApplicationController
 
+  before_action :find_cart
+
   def add
     dish = Dish.find(params[:id])
-    quantity = params[:amount].nil? ? 1 : params[:amount]
-    if session[:cart_id]
-      show
+    if params[:amount].to_i
+      @cart.add(dish, dish.price, params[:amount].to_i)
+      flash[:notice] = "#{dish.name} added to cart: #{params[:amount]}"
     else
-      @cart = Cart.create
-      session[:cart_id] = @cart.id
+      flash[:notice] = "#Please enter a number"
     end
-    @cart.add(dish, dish.price, quantity)
-    flash[:notice] = "#{dish.name} added to cart: #{quantity}"
     redirect_back(fallback_location: (request.referer || restaurant_path(params[:restaurant_id])))
   end
 
+
+  def remove_item
+    dish = Dish.find(params[:dish_id])
+    @cart.remove(dish, 1)
+    flash[:notice] = "#{dish.name} was removed from your cart"
+    redirect_to restaurant_path(params[:restaurant_id])
+  end
+
   def show
     @cart = Cart.find(session[:cart_id])
   end
-end
\ No newline at end of file
+
+  def find_cart
+    if session[:cart_id]
+      @cart = Cart.find(session[:cart_id])
+    else
+      @cart = Cart.create
+      session[:cart_id] = @cart.id
+    end
+  end
```

Comment:
Don't we have a method like this in another controller? If so, perhaps we can move it up in hierarchy to ApplicationController?

Reply:
didnt think of that, will do that.